### ISSUE TITLE
Repeater/ Builder drag&drop

### ISSUE THREAD

**Issue by invaders-xx** [laravel-filament/filament#2141]
### Package

filament/filament

### Package Version

v2.10.44

### Laravel Version

v8.83.8

### Livewire Version

v2.10.5

### PHP Version

PHP 8

### Bug description

I have a JSON structure :
 Repeater field which contains section which contains a builder. Look at the video and the behaviour of dragging block within the builder:

https://user-images.githubusercontent.com/604907/163194474-ed636a80-39d3-49d9-aa74-9532bc7f7de1.mp4

Console  output:
<img width="942" alt="image" src="https://user-images.githubusercontent.com/604907/163195266-fe8d5b43-98a6-45ac-8162-e52160d8fba3.png">

Code of the form:
``` php
public static function getFormSchema(): array
    {
        $aBlocks = [];
        foreach (config('register.types') as $type => $type_name) {
            $aAttr = [];
            $attributes = config('register.attributes.' . $type, []);
            foreach ($attributes as $attribute => $data) {
                switch ($data['type']) {
                    case 'checkbox':
                        $aAttr[] = Forms\Components\Toggle::make($attribute)->label($data['label'] ?? null)->helperText($data['help'] ?? null);
                        break;
                    case 'text':
                        $aAttr[] = Forms\Components\TextInput::make($attribute)->label($data['label'] ?? null)->helperText($data['help'] ?? null);
                        break;
                    case 'number':
                        $aAttr[] = Forms\Components\TextInput::make($attribute)->numeric()->label($data['label'] ?? null)->helperText($data['help'] ?? null);
                        break;
                }
            }
            $aBlocks[] = Forms\Components\Builder\Block::make($type)
                ->label(__($type_name))
                ->schema([
                    Forms\Components\Placeholder::make(__($type_name)),
                    Forms\Components\KeyValue::make('name')
                        ->label(__('Name'))
                        ->keyLabel(__('Language'))
                        ->default(function (Component $livewire) {
                            $locales = [];
                            if ($livewire instanceof Pages\EditRegister) {
                                $locales = $livewire->record->company->languages;
                            }
                            if ($livewire instanceof RegistersRelationManager) {
                                $locales = $livewire->ownerRecord->languages;
                            }
                            return array_fill_keys($locales, null);
                        })
                        ->disableAddingRows()
                        ->disableDeletingRows(),
                    Forms\Components\Section::make('attributes')
                        ->heading(__('Attributes'))
                        ->description(__('Attributes of the field' . ' : ' . __($type_name)))
                        ->schema($aAttr)
                        ->collapsible()
                        ->collapsed(),
                ]);
        }
        return [
            Forms\Components\Grid::make(1)
                ->schema([
                    Forms\Components\Tabs::make('Heading')->tabs([
                        Forms\Components\Tabs\Tab::make(__('General'))->schema([
                            Forms\Components\Group::make()
                                ->schema([
                                    Forms\Components\Card::make()
                                        ->schema([
                                            Forms\Components\Grid::make([
                                                'default' => 1,
                                            ])->schema([
                                                Forms\Components\TextInput::make('name')
                                                    ->label(__('Name'))
                                                    ->required(),
                                                Forms\Components\BelongsToSelect::make('company_id')
                                                    ->label(__('Company'))
                                                    ->searchable()
                                                    ->reactive()
                                                    ->default(function (Component $livewire) {
                                                        if ($livewire instanceof Pages\EditRegister) {
                                                            return $livewire->record->company_id;
                                                        }
                                                        if ($livewire instanceof RegistersRelationManager) {
                                                            return $livewire->ownerRecord->id;
                                                        }
                                                        return null;
                                                    })
                                                    ->disabled(fn(Component $livewire): bool => ($livewire instanceof Pages\EditRegister || $livewire instanceof RegistersRelationManager))
                                                    ->afterStateUpdated(function (Closure $get, Component $livewire) {
                                                        $company_id = $get('company_id');
                                                        if ($company_id) {
                                                            $company = Company::find($company_id);
                                                            $livewire->setTranslatableLocales($company->languages);
                                                        }
                                                    })
                                                    ->relationship('company', 'name', fn(Builder $query) => $query->internal())
                                                    ->required(),
                                            ]),
                                        ]),
                                    Forms\Components\Card::make()
                                        ->schema([
                                            Forms\Components\Grid::make([
                                                'default' => 1,
                                            ])->schema([
                                                Forms\Components\RichEditor::make('description')
                                                    ->label(__('Description'))
                                                    ->hint(__('Translatable'))
                                                    ->hintIcon('heroicon-s-translate'),
                                            ]),
                                        ]),
                                ])->columns([
                                    'sm' => 2,
                                    //'xs' => 1
                                ])->columnSpan([
                                    'sm' => 2,
                                ]),
                            Forms\Components\Card::make()
                                ->schema(static::getRecordInfos())
                                ->columnSpan(1),
                        ])->columns([
                            'sm' => 3,
                            'lg' => null,
                        ]),
                        Forms\Components\Tabs\Tab::make(__('Structure'))
                            ->visible(fn(Component $livewire) => (bool)$livewire->record)
                            ->schema([
                                Forms\Components\Card::make()
                                    ->schema([
                                        Forms\Components\Repeater::make('structure.sections')
                                            ->label(__('Sections'))
                                            ->schema([
                                                Forms\Components\KeyValue::make('name')
                                                    ->label(__('Name'))
                                                    ->keyLabel(__('Language'))
                                                    ->default(function (Component $livewire) {
                                                        $locales = [];
                                                        if ($livewire instanceof Pages\EditRegister) {
                                                            $locales = $livewire->record->company->languages;
                                                        }
                                                        if ($livewire instanceof RegistersRelationManager) {
                                                            $locales = $livewire->ownerRecord->languages;
                                                        }
                                                        return array_fill_keys($locales, null);
                                                    })
                                                    ->disableAddingRows()
                                                    ->disableDeletingRows(),
                                                Forms\Components\Section::make('fields')
                                                    ->heading(__('Fields'))
                                                    ->description(__('List of fields within the section.'))
                                                    ->schema([
                                                        Forms\Components\Builder::make('fields')
                                                            ->blocks($aBlocks)
                                                    ])
                                                    ->collapsible()
                                                    ->collapsed()
                                            ]),
                                    ]),
                            ]),
                    ]),
                ])->columns([
                    'sm' => 1,
                    'lg' => null,
                ])
        ];
    }
```

**Comment by invaders-xx** [comment]
@danharrin any inputs, thoughts ?

**Comment by siegerhansma** [comment]
I'm having a similar issue with a repeater inside a builder. Creating 2 repeater blocks within the builder is fine, but when I drag and drop them I get an exception. I think this is similar to your issue @invaders-xx .

---
Creating the repeater items
![Screenshot 2022-04-14 at 13 40 12](https://user-images.githubusercontent.com/3443339/163383928-446d83f7-41a0-4e38-ae57-b2fce6554290.png)

State at this point
<img width="626" alt="Screenshot 2022-04-14 at 13 40 26" src="https://user-images.githubusercontent.com/3443339/163384022-104042ea-c47f-479c-abee-e88502934bed.png">

After dragging
<img width="550" alt="Screenshot 2022-04-14 at 13 40 52" src="https://user-images.githubusercontent.com/3443339/163384073-bde35cfc-a4ff-4bda-afb0-79942bf69955.png">

Exception that is thrown
<img width="1454" alt="Screenshot 2022-04-14 at 13 41 01" src="https://user-images.githubusercontent.com/3443339/163384116-4efff33f-8fa2-4599-9a7f-01ec33365b39.png">

**Comment by siegerhansma** [comment]
Looking at the server requests, it looks like the wire:sortable event is being triggered on both the builder and repeater because they are nested.
<img width="968" alt="Screenshot 2022-04-14 at 13 58 04" src="https://user-images.githubusercontent.com/3443339/163386612-32ec1ea2-ff1c-48c9-8e7c-700f179dea82.png">

**Comment by danharrin** [comment]
Introduced by #2017, fixed by #2156.

**Comment by danharrin** [comment]
Fixed by #2164.
